File: keybout-backend/src/main/kotlin/com/adeliosys/keybout/model/Word.kt
package com.adeliosys.keybout.model

/**
 * A word or other type or expression (such as a calculus) typed by a user.
 */
class Word(val value: String, val display: String, var userName: String = "") {

    fun conflictsWith(otherValue: String): Boolean = value.startsWith(otherValue) || otherValue.startsWith(value)

    fun copy() = Word(value, display, userName)
}

File: keybout-frontend/src/app/play/word.ts
export class Word {
  label: string;
  userName: string;
  display: string;

  constructor(label: string, userName: string, display: string) {
    this.label = label;
    this.userName = userName;
    this.display = display;
  }

  getEffectiveDisplay() {
    if (this.userName === '') {
      return this.display;
    } else {
      return this.label;
    }
  }
}
